lib/maths/CDecompositionComponent.cc

```diff
@@ -136,6 +137,21 @@ TDoubleDoublePr CDecompositionComponent::value(double offset, double n, double c
             return {m, m};
         }
 
+        // Sanity check that neither m nor sd are NaN values
+        // This is an extremely rare edge case as checks elsewhere prevent NaNs from
+        // being added to the models. As yet it is unknown how NaNs could be returned
+        // from the splines, only that they do occur.
+        // The returned interval will not contribute to the baseline.
+        if (maths::CMathsFuncs::isNan(m)) {
```

Comment:
Is there any chance that `m` could be `nan` when `confidence == 0.0`?  If so then this test would be better done above `if (confidence == 0.0) {` above, to prevent returning `{m, m}` there when `m` is `nan`.